Havit.Blazor.Components.Web.Bootstrap/TreeViews/Internal/HxTreeViewItemInternal.razor

```diff
@@ -2,13 +2,19 @@
 @typeparam TItem
 
 @{
-    var icon = this.IconSelector?.Invoke(this.Item) ?? null;
-    var title = this.TitleSelector?.Invoke(this.Item) ?? null;
-    var cssClassFromSelector = this.CssClassSelector?.Invoke(this.Item) ?? null;
-    var children = this.ChildrenSelector(this.Item);
-    var hasChildren = children?.Any() ?? false;
-    var isSelected = this.Item.Equals(this.TreeViewContainer.SelectedItem);
-    this.IsExpanded ??= this.InitialExpandedSelector?.Invoke(this.Item) ?? false;
+    IconBase icon = this.IconSelector?.Invoke(this.Item) ?? null;
+    string title = this.TitleSelector?.Invoke(this.Item) ?? null;
+    string cssClassFromSelector = this.CssClassSelector?.Invoke(this.Item) ?? null;
+    IEnumerable<TItem> children = this.ChildrenSelector(this.Item);
+    bool hasChildren = children?.Any() ?? false;
+    bool isSelected = this.Item.Equals(this.TreeViewContainer.SelectedItem);
+    bool shouldBeInitiallyExpanded = this.InitialExpandedSelector?.Invoke(this.Item) ?? false;
+
+    if (!hasBeenInitiallyExpanded && shouldBeInitiallyExpanded && hasChildren)
+    {
+        hasBeenInitiallyExpanded = true;
+        IsExpanded = true;
+    }
```

Comment:
Not sure here.
The original implementation was too simple to follow.
The `InitialExpandedSelector` should be used just once in the lifecycle (`OnInitialized`?) and should not be called on every single re-render.